charon/src/bin/charon-driver/translate/translate_constants.rs

```diff
@@ -42,6 +42,16 @@ impl<'tcx, 'ctx, 'ctx1> BodyTransCtx<'tcx, 'ctx, 'ctx1> {
                     }
                 };
                 Literal::Scalar(scalar)
+            },
+            hax::ConstantLiteral::Float(bits, float_type) => {
+                use hax::FloatTy;
+                let value = match float_type {
+                    FloatTy::F16 => FloatValue::F16(*bits as u128),
```

Comment:
why do you need the `as u128`? isn't `*bits` already a `u128`?

Reply:
Not applicable anymore as the representation changed